waggle-dance-core/src/main/java/com/hotels/bdp/waggledance/mapping/model/DatabaseMappingImpl.java

```diff
@@ -205,8 +205,11 @@ public FireEventRequest transformInboundFireEventRequest(FireEventRequest rqst)
 
   @Override
   public ForeignKeysRequest transformInboundForeignKeysRequest(ForeignKeysRequest request) {
-    request.setParent_db_name(metaStoreMapping.transformInboundDatabaseName(request.getParent_db_name()));
-    request.setForeign_db_name(metaStoreMapping.transformInboundDatabaseName(request.getForeign_db_name()));
+    // request.getParent_db_name(); is null, resolve parent_db_name from foreign_db_name using metaStoreMapping
+    String parent_db_name = metaStoreMapping.transformInboundDatabaseName(request.getForeign_db_name());
```

Comment:
we shouldn't reset this to the foreign_db, just leave null if parent_db_name==null otherwise transform